### ISSUE TITLE
Property quarkus.resteasy.gzip.max-input doesn't work

### ISSUE THREAD

**Issue by AShynkevich** [quarkusio/quarkus#12647]
**Describe the bug**
Quarkus doesn't allow to extend uploading file size limitation by property **quarkus.resteasy.gzip.max-input**. Setting the specific value the default size(10M) is still used and I get **413 Request Entity Too Large** error.

**Expected behavior**
Response 201 No content

**Actual behavior**
Response 413 Request Entity Too Large

**To Reproduce**
Link to reproducer: [GitHub](https://github.com/AShynkevich/quarkus-issues)

1. Generated a sinple quarkus project with a resteasy extension. 
2. Set properties on application.properties file:
quarkus.resteasy.gzip.enabled=true
quarkus.resteasy.gzip.max-input=20M
3. Created a simple controller for uploading file:
`@POST
 @Path("/photos")
    public void uploadPhoto(File file) throws IOException {
        System.out.println(file);
   }`
4. Launched application 
5. Tried to upload file (16MB) via Postman 

**Configuration**
```properties
quarkus.resteasy.gzip.enabled=true
quarkus.resteasy.gzip.max-input=20M
```

**Comment by geoand** [comment]
cc @ronsigal @asoldano

**Comment by asoldano** [comment]
@weli, is this perhaps something you can have a look?

**Comment by gsmet** [comment]
I think there's a misunderstanding here: `gzip.max-input` is about the maximum size above which we won't try to gzip the content.

You're probably looking for this instead: https://quarkus.io/guides/http-reference#http-limits-configuration .

**Comment by AShynkevich** [comment]
@gsmet You're right it fixed my issue using property `quarkus.http.limits.max-body-size` and a good guide referrence. Nevertheless, as far as I enabled 'quarkus.resteasy.gzip.enabled=true' doesn't it mean that we're gzipping the content? Maybe, I understand something wrong.

**Comment by AShynkevich** [comment]
@gsmet any comments?

**Comment by gsmet** [comment]
I'm not sure what you're asking.

If you expect what you post to be gzipped, it's the client that should gzip the content, not the server.

**Comment by AShynkevich** [comment]
Ok, thanks a lot for the help! Quarkus is a quite interesting platform. The issue can be closed.

**Comment by StFS** [comment]
This issue is still valid and should not have been closed. Your understanding of the intended behaviour was correct @AShynkevich.

The `quarkus.resteasy.gzip.max-input` property is used in the GZIP filter. The client gzips the upload content and sets the `Content-Encoding: gzip` header and puts the gzipped content in the body. The GZIP filter unzips the request body on the server side and this is where this property should be taken into consideration. If the deflated stream exceeds the value defined in this property, the server should cut off the request and respond with a 413.

This is from the [Quarkus rest-json guide](https://quarkus.io/guides/rest-json#gzip-support):

> Configure the upper limit on deflated request body. This is useful to mitigate potential attacks by limiting their reach.

So the `quarkus.resteasy.gzip.max-input` config parameter should in fact be used to limit the HTTP request body size (after deflation) and it does in fact fail even when you send a >10M request body, even though you configure the config parameter to a larger value than the size of the deflated request.

I don't understand why this issue was closed. I had to create my own GZIP interceptor because of this.
